### DISCUSSION TITLE
Should JFace depend on Commands?

### DISCUSSION THREAD

**Discussion by basilevs** [eclipse-platform/eclipse.platform.ui#1782] [Q&A] [answered]
I'm confused by this dependency:

https://github.com/eclipse-platform/eclipse.platform.ui/blob/c40b7a177e4be437a644ba578ace8ed6d0cbaaf0/bundles/org.eclipse.jface/META-INF/MANIFEST.MF#L38

Should not Eclipse Platform use JFace and not the other way around? Does anyone know/understand the rationale and the effort that might be required to eliminate the dependency?

I've stumbled upon this while trying to create standalone tests/demonstrators/prototypes using JFace without Eclipse.

**Comment by merks** [comment, marked as answer]
I wonder what you consider "Eclipse Platform"? 

Removing the requirement results in hundreds of errors in the project:

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/208716/aaeef63e-3e5b-4566-9964-b148ecdfeb59)

Not to mention the downstream impact of the re-export removal.

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/208716/1ab7cbdc-9141-4772-8bbe-ef7054392d86)

I also don't know what you mean by a JFace without Eclipse.   Since JFace is Eclipse it's hard to know which part of Eclipse you are trying to do without.

In the end, you really do need all these things:

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/208716/8a5ee4ec-5ae6-46e0-9d35-3b8abc663995)

**Reply by basilevs** [reply]
In my case, I was trying to create a program to demonstrate a bug in JFace. I assumed that SWT was its only dependency and it can be used without OSGI.
So from the dependencies on the screenshot, I would expect only `org.eclipse.swt*`.

I admit that I've never inspected the full transitive list, and seeing weaving hook is even more exciting than commands.

This answers my question - JFace was never a standalone UI library and is instead a part of Eclipse Platform. Thanks!

**Reply by merks** [reply]
I'm not sure if JFace can run standalone without an OSGi runtime.  Probably parts of it can.  But you generally would still need the jars on the classpath as implied by the dependencies.
